eos/form-factors/k-lcdas.cc

```diff
@@ -0,0 +1,493 @@
+
+            if (mu < _mu_b)
+                return result * std::pow(alpha_s_mu / alpha_s_c, 1.0 / QCD::beta_function_nf_4[0]);
+
+            double alpha_s_b = model->alpha_s(_mu_b);
+            result *= std::pow(alpha_s_b / alpha_s_c, 1.0 / QCD::beta_function_nf_4[0]);
+
+            if (mu < _mu_t)
+                return result * std::pow(alpha_s_mu / alpha_s_b, 1.0 / QCD::beta_function_nf_5[0]);
+
+            throw InternalError("Implementation<KaonLCDAs>: RGE coefficient must not be evolved above mu_t = " + stringify(_mu_t()));
+        }
+
+        inline double a1K(const double & mu) const
+        {
+            return a1K_0 * std::pow(c_rge(mu), 32.0 / 9.0);
+        }
+
+        inline double a2K(const double & mu) const
+        {
+            return a2K_0 * std::pow(c_rge(mu), 50.0 / 9.0);
+        }
+
+        inline double muK(const double & mu) const
+        {
+            return m_K * m_K / (model->m_s_msbar(mu) + model->m_ud_msbar(mu) / 2.0);
+        }
+
+        double f3K(const double & mu) const
+        {
+            const double c_rge = this->c_rge(mu);
+            const double mu_0  = 1.0; // initial state is fixed at 1 GeV
+            const double m_s_0 = model->m_s_msbar(mu_0);
+
+            return f3K_0 * std::pow(c_rge, 55.0 / 9.0)
+                + 2.0 / 19.0 * (std::pow(c_rge, 4.0)        - std::pow(c_rge, 55.0 / 9.0)) * f_K * m_s_0
+                + 6.0 / 65.0 * (std::pow(c_rge, 55.0 / 9.0) - std::pow(c_rge, 68.0 / 9.0)) * f_K * m_s_0 * a1K_0;
+        }
+
+        double omega3K(const double & mu) const
+        {
+            const double c_rge = this->c_rge(mu);
+            const double mu_0  = 1.0; // initial state is fixed at 1 GeV
+            const double m_s_0 = model->m_s_msbar(mu_0);
+
+            return f3K_0 * omega3K_0 * std::pow(c_rge, 105.0 / 9.0)
+                + 1.0 / 170.0 * (std::pow(c_rge, 4.0)        - std::pow(c_rge, 104.0 / 9.0)) * f_K * m_s_0
+                + 1.0 /  10.0 * (std::pow(c_rge, 68.0 / 9.0) - std::pow(c_rge, 104.0 / 9.0)) * f_K * m_s_0 * a1K_0
+                + 2.0 /  15.0 * (std::pow(c_rge, 86.0 / 9.0) - std::pow(c_rge, 104.0 / 9.0)) * f_K * m_s_0 * a2K_0;
+        }
+
+        double lambda3K(const double & mu) const
+        {
+            const double c_rge = this->c_rge(mu);
+            const double mu_0  = 1.0; // initial state is fixed at 1 GeV
+            const double m_s_0 = model->m_s_msbar(mu_0);
+
+            return f3K_0 * lambda3K_0 * std::pow(c_rge, 139.0 / 18.0)
+                - 14.0 / 67.0 * (std::pow(c_rge, 4.0)        - std::pow(c_rge, 139.0 / 18.0)) * f_K * m_s_0
+                + 14.0 /  5.0 * (std::pow(c_rge, 68.0 / 9.0) - std::pow(c_rge, 139.0 / 18.0)) * f_K * m_s_0 * a1K_0
+                - 4.0  / 11.0 * (std::pow(c_rge, 86.0 / 9.0) - std::pow(c_rge, 139.0 / 18.0)) * f_K * m_s_0 * a2K_0;
+        }
+
+        inline double eta3K(const double & mu) const
+        {
+            return f3K(mu) / (f_K() * muK(mu));
```

Comment:
I do not understand why there are paranthesis for the "f_K()" here. Can someone explain this to me?

Reply:
The type of the object named ``f_K`` is ``UsedParameter``. This class provides two operators to obtain the parameter's value:
 - a conversion operator to ``double``, which is triggered implicitly by using the parameter in place of a ``double`` anywhere, or explicitly by using ``static_cast<double>``; and
 - a function call operator, which is triggered by treating the object like a function and call it (without any arguments).

Reply:
https://github.com/eos/eos/blob/19d11d6be8042a1e279146388109a3e0002f7b0a/eos/utils/parameters.hh#L247-L251